ghost/core/core/server/data/importer/import-manager.js

```diff
@@ -351,45 +335,132 @@ class ImportManager {
     /**
      * Import Step 4:
      * Report on what was imported, currently a no-op
-     * @param {ImportResult[]} importResults
-     * @returns {Promise<ImportResult[]>} importResults
+     * @param {Object.<string, ImportResult>} importResults
+     * @returns {Promise<Object.<string, ImportResult>>} importResults
      */
     async generateReport(importResults) {
         return Promise.resolve(importResults);
     }
 
+    /**
+     * Step 5:
+     * Remove files after we're done (abstracted into a function for easier testing)
+     * @returns {Promise<void>}
+     */
+    async cleanUp() {
+        if (this.fileToDelete === null) {
+            return;
+        }
+
+        try {
+            await fs.remove(this.fileToDelete);
+        } catch (err) {
+            logging.error(new errors.InternalServerError({
+                err: err,
+                context: tpl(messages.couldNotCleanUpFile.error),
+                help: tpl(messages.couldNotCleanUpFile.context)
+            }));
+        }
+
+        this.fileToDelete = null;
+    }
+
+    /**
+     * Import Step 6:
+     * Create an email to notify the user that the import has completed
+     * @param {ImportResult} result
+     * @param {Object} options
+     * @param {string} options.emailRecipient
+     * @param {string} options.importTag
+     * @returns {string}
+     */
+    generateCompletionEmail(result, {
+        emailRecipient,
+        importTag
+    }) {
+        const siteUrl = new URL(urlUtils.urlFor('home', null, true));
+        const postsUrl = new URL('posts', urlUtils.urlFor('admin', null, true));
+        if (importTag && result?.data?.tags) {
+            const tag = result.data.tags.find(t => t.name === importTag);
+            postsUrl.searchParams.set('tag', tag.slug);
+        }
+
+        return emailTemplate({
+            result,
+            siteUrl,
+            postsUrl,
+            emailRecipient
+        });
+    }
+
     /**
      * Import From File
      * The main method of the ImportManager, call this to kick everything off!
      * @param {File} file
      * @param {ImportOptions} importOptions to allow override of certain import features such as locking a user
-     * @returns {Promise<ImportResult[]>}
+     * @returns {Promise<Object.<string, ImportResult>>}
      */
     async importFromFile(file, importOptions = {}) {
+        if (!importOptions.forceInline && !importOptions.runningInJob) {
```

Comment:
I think instead of "forceInline" we could use the pattern of checking for test environment `env?.startsWith('testing')` (l[ike done here](https://github.com/TryGhost/Ghost/blob/9e35c4dd7a8c1904a5162bc064db2b3e04757ef5/ghost/core/core/server/services/members/service.js#L150)). No need to expose extra parameters in the API if it does not have a valid usecase.

Reply:
> I think instead of "forceInline" we could use the pattern of checking for test environment `env?.startsWith('testing')` (l[ike done here](https://github.com/TryGhost/Ghost/blob/9e35c4dd7a8c1904a5162bc064db2b3e04757ef5/ghost/core/core/server/services/members/service.js#L150)). No need to expose extra parameters in the API if it does not have a valid usecase.

Totally! I think I copied the `forceInline` out of the members importer but using env sounds perfect. Anything to get the tests passing :)